projects/aca-content/src/lib/components/common/toggle-shared/toggle-shared.component.ts

```diff
@@ -22,48 +22,61 @@
  * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Component, OnInit, Input } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
 import { Store } from '@ngrx/store';
 import { SelectionState } from '@alfresco/adf-extensions';
 import { AppStore, ShareNodeAction, getAppSelection } from '@alfresco/aca-shared/store';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-toggle-shared',
   templateUrl: './toggle-shared.component.html'
 })
-export class ToggleSharedComponent implements OnInit {
+export class ToggleSharedComponent implements OnInit, OnDestroy {
   @Input()
   data: {
     iconButton?: string;
   };
 
   selection$: Observable<SelectionState>;
+  selectionState: SelectionState;
+  selectionLabel = '';
+  isShared = false;
+
+  onDestroy$ = new Subject<boolean>();
```

Comment:
It is enough to have void instead of boolean because you even do not use false.

Reply:
Removed boolean from subject. New declaration is 

`onDestroy$ = new Subject();`

Reply:
@swapnil-verma-gl: 
new Subject() is not the same like suggested by me new Subject<void>(); 

The reason is that Subject takes 'unknown' type in <> as default if you don't specific anything, not void. So that causes a problem that now we can emit anything but we don't need emit anything really for example: 

<img width="327" alt="image" src="https://github.com/Alfresco/alfresco-content-app/assets/115619721/7ad597e6-8a70-4624-8d08-154dc824f6e2">

You see I can emit empty but also for example with numbers. But if I change it to void instead: 
<img width="903" alt="image" src="https://github.com/Alfresco/alfresco-content-app/assets/115619721/f7ede5de-3c33-4cd1-b80c-c66fd84c6af6">


Then it does not allow me to emit any value. Please use void in <> as having default uknown is redundant.

Reply:
Oh, ok. I was not aware of that. I have made the necessary changes. New declaration is 
`onDestroy$ = new Subject<void>();`